racket/src/racket/src/schpriv.h

```diff
@@ -1459,6 +1459,104 @@ Scheme_Object *scheme_top_introduce(Scheme_Object *form, Scheme_Env *genv);
 /*                   syntax run-time structures                           */
 /*========================================================================*/
 
+/* A Scheme_Compiled_Local record represents a local variable,
+   both the binding and references to that binding. When inlining
+   of other transformations duplicate a variable, a new instance
+   is allocated to represent a separate variable. Different passes
+   in the comiler store different information about the variable. */
+typedef struct Scheme_Compiled_Local
+{
+  Scheme_Object so;
+
+  /* The `mode` determines which of the union cases below
+     is active, corresponding to information for a particular
+     pass: */
+  unsigned int mode : 3;
```

Comment:
Does `mode` take on values from the `#define`s below (such as `SCHEME_VAR_MODE_LETREC_CHECK`)? If so, maybe a pointer there would be a good addition to this comment, but I'm confused about why there are more than 3 of those modes, since there are 3 cases in the union.